components/org.wso2.carbon.identity.notification.push.provider/src/main/java/org/wso2/carbon/identity/notification/push/provider/model/PushNotificationData.java

```diff
@@ -0,0 +1,208 @@
+package org.wso2.carbon.identity.notification.push.provider.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Model class for push notification data.
+ */
+public class PushNotificationData {
+
+    private String notificationTitle;
+    private String notificationBody;
+    private String username;
+    private String tenantDomain;
+    private String userStoreDomain;
+    private String applicationName;
+    private String notificationScenario;
+    private String pushId;
+    private String deviceToken;
+    private String challenge;
```

Comment:
Improvement: lets only have the generic attributes and have a map to store any context specific data such as the challenge used when doing push based authentication